src/features/plan/components/CreateBudgetForm.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from "react";
+import * as Yup from "yup";
+import { useDispatch } from "react-redux";
+import { format, isMatch } from "date-fns";
+import FormTextField from "../../../common/components/form/FormTextField";
+import FormModal from "../../../common/components/form/FormModal";
+import { createBudget, createStartDate, createEndDate } from "../state/budgets";
+
+export default function CreateBudgetForm({ afterSubmit }) {
+    const dispatch = useDispatch();
+    const today = format(new Date(), "dd/MM/yyyy");
+    const [startDate, setStartDate] = useState("");
+    const [endDate, setEndDate] = useState("");
+
+    return (
+        <FormModal
+            title="Set your budget..."
+            heading="Please set a start date, end date and budget amount for
+                        your upcoming budget."
+            submitText="Create"
```

Comment:
In general it's more user-friendly to provide context to action buttons - "Create budget" here is more helpful.